### PR TITLE
SUBMARINE-1238. Create environment of the React workbench

### PR DESCRIPTION
**Pull request by KUAN-HSUN-LI**
### What is this PR for?
Create the environment for the React workbench.
Why do we need the new workbench?
1. The old workbench with Angular is not efficient
2. API service in the old workbench is too complicated. We will generate the API via `openAPI` in the new workbench.
3. The old workbench is not fully test

### What type of PR is it?
[Feature]

### Todos
* [x] - Build the code from `vite-app` template
* [x] - Create unit test via `react testing library`
* [x] - Let sidebar collapsible

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-1238

### How should this be tested?
* The test has been added to the GitHub Actions.
* You can also run the test in the `submarine-workbench-v2` through the command `yarn test`.

### Screenshots (if appropriate)
image

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? Yes
* Does this need new documentation? No

### CODE REVIEW

**Comment by KUAN-HSUN-LI** [comment]
@jeff-901 @featherchen @noidname01 Help me review the code. Thanks in advance!

**Comment by jeff-901** [comment]
> @jeff-901 @featherchen @noidname01 Help me review the code. Thanks in advance!

I cannot install the package because of the dependency. Is this the problem only for M1?
image

**Comment by KUAN-HSUN-LI** [comment]
> > @jeff-901 @featherchen @noidname01 Help me review the code. Thanks in advance!
> 
> I cannot install the package because of the dependency. Is this the problem only for M1? image

Maybe I should downgrade the React from 18.0.0 to 17.0.2

**Comment by noidname01** [comment]
LGTM!

File: submarine-workbench-v2/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+## How to develop
+```
+npm run dev --open (or yarn dev --open)
+```
+
+## How to Build
+```
+npm build (or yarn build)
```
Comment (jeff-901) [inline submarine-workbench-v2/README.md:31]: Should be `npm run build`

File: submarine-workbench-v2/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+coverage/
```
Comment (jeff-901) [inline submarine-workbench-v2/.gitignore:1]: Do we need to add `package-lock.json` and `yarn.lock`?

**Review by jeff-901 — APPROVED** [review]

File: submarine-workbench-v2/.gitignore (hunk 2 above)
Comment (KUAN-HSUN-LI) [reply submarine-workbench-v2/.gitignore:1]: I think the lock file is required

**Pull request closed without merging**